app/models/github_user.rb

```diff
@@ -44,6 +44,9 @@ def create_secret
   end
 
   def prs_for_review_by_status(status)
-    reviews.where(status: status).map(&:pull_request)
+    reviews.where(status: status)
```

Comment:
I'm not sure this is entirely better, but it could be expressed as a relation, which would get cached once called

```ruby
class GithubUser < AR::Base
  has_many :watched_pull_requests, ->(status){ where(status: :open, review: { status: status }) }, class_name: "PullRequest", through: :reviews
end
```